internal/bzlmod/go_deps.bzl

```diff
@@ -602,6 +603,10 @@ def _go_deps_impl(module_ctx):
             # Do not create a go_repository for a dep shared with the non-isolated instance of
             # go_deps.
             continue
+        if module.repo_name in repos_processed:
+            fail("Go module {prev_path} and {path} will resolve to the same Bazel repo name: {name}. While Go allows modules to only differ in case, this isn't supported in Gazelle (yet). Please ensure you only use one of these modules in your go.mod(s)")
```

Comment:
This is not an f-string, you need a call to `.format`.

Reply:
Oops, my bad. Fixed: https://github.com/bazelbuild/bazel-gazelle/pull/1954/commits/63ee5e2f57537fef1401693a6c8b17f4ad1d7e77